mouhamadouBarry/assignments/treesBFS/lc102/lc102.js

```diff
@@ -1,7 +1,38 @@
 // Problem Statement #
 
-// Given a binary tree, populate an array to represent its level-by-level traversal. You should populate the values of all nodes of each level from left to right in separate sub-arrays.
+// Given a binary tree, populate an array to represent its level-by-level traversal. 
+// You should populate the values of all nodes of each level from left to right in separate sub-arrays.
 
+/*-----------------Initial Thought----------------------
+Input: A Binary Tree
+Output: Level by level representation
+
+Initializations:
+  outputArray - an array that will store the results
+  currentlevel - queue.shift
+
+Logic:
+  
+while(queue)
+  level queue =[]
+  currentlevel - queue.shift  
+
+for (i=0; i< ( something ); i++)
+  if(currentlevel.left)
+    levelqueue.push( currentlevel.left )
+  if(currentlevel.right)
+    levelqueue.push( currentlevel.right )
+
+-output queue
+
+[12,7,1,9,10,5]
+
```

Comment:
Good psuedocode. For an approach, it will be more helpful to have it written out in sentences in a high-level manner. That way regardless of the language being used, the interviewer can understand your intention with your code.